orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/model/Deployments.kt

```diff
@@ -56,22 +62,64 @@ internal data class Baseline(
 )
 
 /**
- * The deployment context for a single cluster.
+ * A typed subset of the full cluster context.
  */
 internal data class ClusterSpec(
   val cloudProvider: String,
   val account: String,
-  val moniker: Moniker,
+  val moniker: Moniker?,
   val availabilityZones: Map<String, Set<String>>,
-  val region: String?
+  val region: String?,
+  val application: String?,
+  val stack: String?,
+  val freeFormDetails: String?
```

Comment:
Forgive my ignorance here, but why add these additional values if we've got moniker? Shouldn't we just try to converge around that instead of separate properties?

Reply:
We should be converging - but looks like most of the providers don't support moniker. Deck and Clouddriver don't support monikers for either EC2 or GCE (for deploy ops, I mean).